### PR TITLE
DATA: Adding PostReport Model + Brokers

### PR DESCRIPTION
**Pull request by imanvt**

### CODE REVIEW

**Comment by imanvt** [comment]
#177

File: Taarafo.Core/Brokers/Storages/StorageBroker.PostReports.cs [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+//
+using Microsoft.EntityFrameworkCore;
+using Taarafo.Core.Models.Posts;
+
+namespace Taarafo.Core.Brokers.Storages
+{
+    public partial class StorageBroker
+    {
+		public DbSet<PostReport> PostReports { get; set; }
+	}
```
Comment (hassanhabib) [inline Taarafo.Core/Brokers/Storages/StorageBroker.PostReports.cs:13]: Please fix the indentations in this file.
So the closing squiggly brackets need to be at the same level as the opening ones.
And the data set property needs to be one tab after that.

**Comment by hassanhabib** [comment]
@imanvt please set the relationships explicitly so we can define other conditions.

check out this for reference:
https://github.com/hassanhabib/CulDeSacDemoApi/pull/2/files#diff-dbe5b6959b8c4d11622b109be33ff751cc6847eb9f977adeb16cdc50bacc4aa3

File: Taarafo.Core/Migrations/20220324042442_AddPostReportModel.cs [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+        protected override void Up(MigrationBuilder migrationBuilder)
+        {
+
+        }
+
+        protected override void Down(MigrationBuilder migrationBuilder)
+        {
+
+        }
+    }
```
Comment (hassanhabib) [inline Taarafo.Core/Migrations/20220324042442_AddPostReportModel.cs:18]: This file here is not needed. it just means you ran migrations with no changes. You can delete this guy

File: Taarafo.Core/Brokers/Storages/StorageBroker.PostReports.References.cs [hunk 3]
```diff
@@ -0,0 +1,21 @@
+﻿// ---------------------------------------------------------------
+// Copyright (c) Coalition of the Good-Hearted Engineers
+// FREE TO USE TO CONNECT THE WORLD
+//
```
Comment (hassanhabib) [inline Taarafo.Core/Brokers/Storages/StorageBroker.PostReports.References.cs:4]: We need to fix this guy with the ```-----------------------------------------```
We also need a new line between the copyrights notice and the code

**Review by ElbekDeveloper — APPROVED** [review]

File: Taarafo.Core/Models/Profiles/Profile.cs [hunk 4]
```diff
@@ -15,5 +17,8 @@ public class Profile
         public string Email { get; set; }
         public DateTimeOffset CreatedDate { get; set; }
         public DateTimeOffset UpdatedDate { get; set; }
+
+        public IEnumerable<PostReport> PostReports { get; set; }
```
Comment (ShriHumrudha) [inline Taarafo.Core/Models/Profiles/Profile.cs:21]: How about a meaningful name ReportedPosts?
Remove the extra line at 22

File: Taarafo.Core/Models/Posts/Post.cs [hunk 5]
```diff
@@ -18,5 +18,7 @@ public class Post
         public DateTimeOffset UpdatedDate { get; set; }
 
         public IEnumerable<Comment> Comments { get; set; }
+        public IEnumerable<PostReport> PostReports { get; set; }
+
```
Comment (ShriHumrudha) [inline Taarafo.Core/Models/Posts/Post.cs:22]: Extra line can be removed

File: Taarafo.Core/Brokers/Storages/StorageBroker.PostReports.References.cs [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+// ---------------------------------------------------------------
+
+using Microsoft.EntityFrameworkCore;
+using Taarafo.Core.Models.Posts;
+
+namespace Taarafo.Core.Brokers.Storages
+{
+    public partial class StorageBroker
+    {
+        private static void AddPostReportsReferences(ModelBuilder modelBuilder)
```
Comment (ShriHumrudha) [inline Taarafo.Core/Brokers/Storages/StorageBroker.PostReports.References.cs:13]: This method has to be called on migrating.

**Review by ShriHumrudha — CHANGES_REQUESTED** [review]

File: Taarafo.Core/Brokers/Storages/StorageBroker.PostReports.References.cs (hunk 6 above)
Comment (imanvt) [reply Taarafo.Core/Brokers/Storages/StorageBroker.PostReports.References.cs]: Ran this command: Add-Migration AddPostReportsReferences
Is this what you meant?

File: Taarafo.Core/Models/Posts/PostReport.cs [hunk 7]
```diff
@@ -0,0 +1,25 @@
+﻿// ---------------------------------------------------------------
```
Comment (ShriHumrudha) [inline Taarafo.Core/Models/Posts/PostReport.cs:1]: Should this more go under 'PostReport' folder?

File: Taarafo.Core/Models/Profiles/Profile.cs [hunk 8]
```diff
@@ -15,5 +17,8 @@ public class Profile
         public string Email { get; set; }
         public DateTimeOffset CreatedDate { get; set; }
         public DateTimeOffset UpdatedDate { get; set; }
+
+        public IEnumerable<PostReport> ReportedPosts { get; set; }
+
```
Comment (ShriHumrudha) [inline Taarafo.Core/Models/Profiles/Profile.cs:22]: Remove this extra line

File: Taarafo.Core/Models/Posts/Post.cs [hunk 9]
```diff
@@ -18,5 +18,6 @@ public class Post
         public DateTimeOffset UpdatedDate { get; set; }
 
         public IEnumerable<Comment> Comments { get; set; }
+        public IEnumerable<PostReport> ReportedPosts { get; set; }
```
Comment (ShriHumrudha) [inline Taarafo.Core/Models/Posts/Post.cs:21]: How about 'PostReported'?

File: Taarafo.Core/Brokers/Storages/StorageBroker.PostReports.References.cs [hunk 10]
```diff
@@ -0,0 +1,28 @@
...
+// ---------------------------------------------------------------
+
+using Microsoft.EntityFrameworkCore;
+using Taarafo.Core.Models.Posts;
+
+namespace Taarafo.Core.Brokers.Storages
+{
+    public partial class StorageBroker
+    {
+        private static void AddReportedPostsReferences(ModelBuilder modelBuilder)
```
Comment (ShriHumrudha) [inline Taarafo.Core/Brokers/Storages/StorageBroker.PostReports.References.cs:13]: Tha name can be aligned with the entity name - PostReportsReferences

**Review by ShriHumrudha — CHANGES_REQUESTED** [review]

**Comment by ShriHumrudha** [comment]
@imanvt Need your attention here. The PR will be closed in 3 days.

**Comment by imanvt** [comment]
@ShriHumrudha I resolved all the requested changes I think... am I missing something?

**Comment by ShriHumrudha** [comment]
> @ShriHumrudha I resolved all the requested changes I think... am I missing something?

image

**Comment by ShriHumrudha** [comment]
Closing due to inactivity.

**Pull request closed without merging**